espnet/nets/pytorch_backend/e2e_asr_transformer.py

```diff
@@ -160,6 +162,44 @@ def reset_parameters(self, args):
         # initialize parameters
         initialize(self, args.transformer_init)
 
+    def wfst_based_decoding(self, x, HLG, recog_args, char_list=None):
```

Comment:
Instead of adding it under transformer, I think you can add it in https://github.com/espnet/espnet/blob/master/espnet/nets/pytorch_backend/e2e_asr.py
Then, you can use it for all our architectures.

Probably you can move this method in addition to 
https://github.com/espnet/espnet/blob/c744073fc66328d27642f0083514a3a3625c677b/espnet/nets/pytorch_backend/e2e_asr.py#L379-L399

@ShigekiKarita and @YosukeHiguchi, could you give us some comments?

Reply:
@YosukeHiguchi, do you think we should make it similar to your mask CTC inference, or my above idea would be OK?

Reply:
If my understanding is correct, E2E classes in e2e_asr.py and e2e_asr_transformer.py are defined independently of each other. So this method should be added in both classes so that RNN-based models from e2e_asr.py and Transformer-based models from e2e_asr_transformer.py can use it.